library/libwaku.nim

```diff
@@ -658,5 +698,21 @@ proc waku_stop_discv5(
 
   return RET_OK
 
+proc waku_peer_exchange_request(
+    ctx: ptr WakuContext, numPeers: uint64, callback: WakuCallBack, userData: pointer
+): cint {.dynlib, exportc.} =
+  checkLibwakuParams(ctx, callback, userData)
+
+  let discoveredPeers = waku_thread.sendRequestToWakuThread(
+    ctx, RequestType.DISCOVERY, DiscoveryRequest.createPeerExchangeRequest(numPeers)
+  ).valueOr:
+    let msg = $error
+    callback(RET_ERR, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+    return RET_ERR
+
+  let msg = $discoveredPeers
+  callback(RET_OK, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+  return RET_OK
```

Comment:
Can't this repeated part be abstracted?

Reply:
Yup, I will double-check that in a separate PR